modules/status-roles/events/presenceUpdate.js

```diff
@@ -0,0 +1,22 @@
+const {localize} = require("../../../src/functions/localize");
+
+module.exports.run = async function (client, newPresence) {
+
+    if (!client.botReadyAt) return;
+    const moduleConfig = client.configurations['status-roles']['config'];
+    const roles = moduleConfig.roles;
+    const status = moduleConfig.words;
+    const member = newPresence.member;
+
+    if (newPresence.activities.length > 0) {
+        if (status.some(word => newPresence.activities[0].state.includes(word))) {
+            return member.roles.add(roles, localize('mass-role', 'fulfilled'));
+        } else {
+            for (let i = 0; i < roles.length; i++) {
```

Comment:
Note: You could simplify this by using

- `for (const role of roles)`
- `for (const key in roles)

(better readability)

Reply:
ok. i understand, that this is posible. but how do I get the i for checking and removing all roles then? If I just write the i++ into the loop it would be almost the same as it is now

Reply:
`roles` is an array with objects inside them. You can get an object at a position by using `roles[i]` (this is what you do now). 

- With `for (const role of roles)` you get the object without needing to query it yourself. (`role` would be equal to `roles[i]`)
- WIth `for (const key in roles)` you loop though each key (corosponding to position in arrays). In your case, the `i` would be `key`. (You could also just write `for (const i in roles)`

Note that this was only a note for better readability (dx) and can be ignored, if you don't want to include this change. Press "Resolve conversation" below to do so.

Reply:
I already included it. It is always interesting to learn new stuff (as long as it isn't latin)

Reply:
i agree.